### ISSUE TITLE
Replace doughnut-chart with flexible pie-chart

### ISSUE THREAD

**Issue by Joelius300** [Joelius300/ChartJSBlazor#40]
In ~~https://github.com/Joelius300/ChartJSBlazor/pull/30~~ https://github.com/Joelius300/ChartJSBlazor/pull/23 we wanted to remove the Doughnut-stuff entirely and talked about it quite a bit.  
We didn't do it because @SeppPenner said that it could bring us issues and we weren't sure what the best approach was.  

I have now created a [proposal branch](https://github.com/Joelius300/ChartJSBlazor/tree/pie-proposal) where you can see what I want.   
Also [here](https://github.com/Joelius300/ChartJSBlazor/tree/doughnut-pie_comparison) is a comparison branch where you can see that there's no difference between an actual doughnut-chart and a pie-chart with `CutoutPercent` set  to 50.  
Now I think some people thought it's an issue that we use `ChartType.Pie` for a doughnut-chart but that's in fact absolutely fine.  
The only thing this influences is that they have a single different default value `CutoutPercent`. This property defines how much percent of the radius is empty inside making it look more like a doughnut or more like a pie respectively.  
The default value for `Pie` is 0 and 50 for `Doughnut`. They are registered under aliases and it's confirmed that they don't have any other differences. For this reason we can ues `ChartType.Pie` for a doughnut-chart without any concerns.  

Here's a snippet from the chart.js-docs that emphazises this (from [here](https://www.chartjs.org/docs/latest/charts/doughnut.html)): 

> Pie and doughnut charts are effectively the same class in Chart.js, but have one different default value - their cutoutPercentage. This equates what percentage of the inner should be cut out. This defaults to 0 for pie charts, and 50 for doughnuts.

> They are also registered under two aliases in the Chart core. Other than their different default value, and different alias, they are exactly the same.

If noone has any "objections" or sees an issue with that, I will merge [that branch](https://github.com/Joelius300/ChartJSBlazor/tree/pie-proposal) into master, removing `Doughnut`-stuff entirely.  
I will of course document somewhere (in the Wiki, btw we really need to add more stuff there) how you can use `Doughnut`-chart now.  

I'd like approval of @SeppPenner before doing this.

**Comment by SeppPenner** [comment]
Yes, it definitely brings issues because the chart type is set in the `PieChartConfig` (see https://github.com/Joelius300/ChartJSBlazor/issues/31#issuecomment-519455284). This would mean that we had only one configuration file, e.g. `PieChartConfig`, but need to address pie and doughnut chart somehow with a generic way. I had no idea how to do that...

> I have now created a proposal branch where you can see what I want.
Also here is a comparison branch where you can see that there's no difference between an actual doughnut-chart and a pie-chart with CutoutPercent set to 50.

I will take a look at this.

> For this reason we can ues ChartType.Pie for a doughnut-chart without any concerns.

Seriously? I have tried this today and this totally messed up the chart... I will try this again tomorrow, but maybe I did something wrong...

> I'd like approval of @SeppPenner before doing this.

Sure, I have a local fork of the project in a productive project I use (and will try to apply what you just told me tomorrow).

**Comment by SeppPenner** [comment]
Oh, I have an idea... Maybe, I used the `Circumference` value without `Math.Pi` and this messed up the chart. I will try this tomorrow.

**Comment by SeppPenner** [comment]
Another hint: I guess, we don't need the label (Check https://jsfiddle.net/38shmLt1/3/) in the pie and doughnut chart. This doesn't change anything.

The chart type is needed, however. Check https://github.com/Joelius300/ChartJSBlazor/issues/31: The chart type on dataset level is needed for mixed datasets. While you apply the changes, please also approve that you're changing this already.

**Comment by Joelius300** [comment]
> Seriously? 

Yes, I'm almost certain that there is no difference. Again, in the official chart.js docs they literally say that it's an _alias_ at the core and only affects a single default value ([source](https://www.chartjs.org/docs/latest/charts/doughnut.html)). Because of this very small difference, e don't need to "respect" that, we can just use `Pie` and set the different default value ourselfs.  

> I will try this tomorrow.

Thanks. I have not been able to reproduce your issue and after examining the chart.js docs and some more stuff I also don't think I should be able to.  
If you still have problems, please share with every little detail because I really think it would be best if we remove all the doughnut stuff (regarding confusion, size, maintainabilty, redundancy, etc).

>  I guess, we don't need the label (Check https://jsfiddle.net/38shmLt1/3/) in the pie and doughnut chart. This doesn't change anything.

As far as I can see `Label` has been removed from `PieChartDataset` and `DoughnutChartDataset` and is currently not in the master branch. So that's probably done.

> The chart type is needed, however.

We leave that in either way. Sometimes it's needed (mixed sets), other times it's not. It doesn't cause any issues if it's set unnecessarily so we don't need to remove it.

**Comment by SeppPenner** [comment]
> Yes, I'm almost certain that there is no difference.

You're right. Check https://jsfiddle.net/g1anoety/. It works with `pie` as well. I guess, we can finally kill the undead `doughnut chart` then ;)

I have tried this with the Blazor component as well. There is no difference.

> As far as I can see Label has been removed from PieChartDataset and DoughnutChartDataset and is currently not in the master branch. So that's probably done.

Yeah, for the pie and doughnut. But what about the others? We need to investigate if it's needed and if yes where. I haven't seen this working properly anywhere (Just to make it clear again: We're talking about a label on dataset basis. In my opinion, this is only possible with the datalabels plugin for all charts). --> I created a seperate issue here: https://github.com/Joelius300/ChartJSBlazor/issues/46.

> We leave that in either way. Sometimes it's needed (mixed sets), other times it's not. It doesn't cause any issues if it's set unnecessarily so we don't need to remove it.

I agree on that and can verify this. I have already tried this.

**Comment by Joelius300** [comment]
https://github.com/Joelius300/ChartJSBlazor/pull/47 has been merged and all Doughnut-stuff was removed.  

I will keep this issue open as a reminder to update the docs (readme and wiki).
